docs/madr/decisions/006-mesh-traffic-permission.md

```diff
@@ -0,0 +1,785 @@
+# MeshTrafficPermission
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4222
+
+## Context and Problem Statement
+
+[New policy matching MADR](./005-policy-matching.md) introduces a new approach how Kuma applies configuration to proxies.
+Rolling out strategy implies creating a new set of policies that satisfy a new policy matching.
+Current MADR aims to define a MeshTrafficPermission.
+
+## Considered Options
+
+* Create a MeshTrafficPermission policy
+
+## Decision Outcome
+
+Chosen option: create a MeshTrafficPermission policy
+
+### Overview
+
+MeshTrafficPermission allows blocking unwanted incoming traffic for the service.
+There are might be several reasons why it could be useful to block unwanted traffic:
+* following the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege)
+* cut off access for clients that shouldn't use your service (legacy clients, clients from another zone, etc.)
+
+### Specification
+
+Top-level targetRef can have all available kinds:
+
+```yaml
+targetRef:
+  kind: Mesh|MeshSubset|MeshService|MeshServiceSubset|MeshGatewayRoute|MeshHTTPRoute
+  name: ...
+```
+
+MeshTrafficPermission is an inbound policy that's why it has a "from" array:
+
+```yaml
+from:
+  - targetRef:
+      kind: Mesh|MeshSubset|MeshService|MeshServiceSubset
+      name: ...
+```
+
+Configuration for MeshTrafficPermission has a single field "action" with values "ALLOW", "DENY", 
+"ALLOW_WITH_SHADOW_DENY" and "DENY_WITH_SHADOW_ALLOW".
+So the overall specification looks like this:
+
+```yaml
+type: MeshTrafficPermission
+mesh: default
+spec:
+  targetRef:
+    kind: Mesh|MeshSubset|MeshService|MeshServiceSubset|MeshGatewayRoute|MeshHTTPRoute
+    name: ...
+  from:
+    - targetRef:
+        kind: Mesh|MeshSubset|MeshService|MeshServiceSubset
+        name: ...
+      default:
+        action: ALLOW|DENY|ALLOW_WITH_SHADOW_DENY|DENY_WITH_SHADOW_ALLOW
+```
+
+We don't want to mix mTLS configs (like permissive mTLS or TLS version) with MeshTrafficPermission. 
+The main reason is that mTLS settings are applied for the entire inbound, while permissions could be set for a group of clients. 
+
+### Shadow actions
+
+Shadow actions are not enforced but emit stats and can be used for rule testing.
+For example, before creating "action: DENY" rule, user creates "action: ALLOW_WITH_SHADOW_DENY" rule and checks the stats.
+If there are no unexpected denied requests from the clients then it's safe to change "action" to "DENY".
```

Comment:
Comment: If there are, what does the user do then? For this feature to be useful it needs to be coupled with `access_log_hint` (+ maybe some other mechanisms) so that the user can identify the caller.

Reply:
Yeah, right now, it's just an indication some clients request the service. Users should set access logs or check application logs to figure out who are these clients